From f524e739dd117beb239c5299d2dced7a627a8ccc Mon Sep 17 00:00:00 2001
From: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Date: Mon, 26 Nov 2018 12:45:57 +0900
Subject: [PATCH 540/909] arm64: dts: r8a77990-ebisu: Enable simultaneous
 output of VGA and HDMI

commit d3394635755093f8539acfdbaf932a164ec59533 from
git://git.kernel.org/pub/scm/linux/kernel/git/horms/renesas-bsp.git

Current definition is only HDMI single output via DU0.
This patch can simultaneously output HDMI and VGA by adding definition to
VGA via DU1.

x13_clk is detached from the clocks property of the LVDS1 node.
This is because VGA does not display if x13_clkis defined in the clocks
property of the LVDS1 node when simultaneously outputting VGA and HDMI.

Signed-off-by: Takeshi Kihara <takeshi.kihara.df@renesas.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../arm64/boot/dts/renesas/r8a77990-ebisu.dts | 24 +++++++++----------
 1 file changed, 12 insertions(+), 12 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/r8a77990-ebisu.dts b/arch/arm64/boot/dts/renesas/r8a77990-ebisu.dts
index 24090239179a..dbed1018c5b9 100644
--- a/arch/arm64/boot/dts/renesas/r8a77990-ebisu.dts
+++ b/arch/arm64/boot/dts/renesas/r8a77990-ebisu.dts
@@ -247,7 +247,7 @@
 			port@0 {
 				reg = <0>;
 				adv7123_in: endpoint {
-					remote-endpoint = <&du_out_rgb>;
+					remote-endpoint = <&lvds1_out>;
 				};
 			};
 			port@1 {
@@ -289,7 +289,6 @@
 		#clock-cells = <0>;
 		clock-frequency = <74250000>;
 	};
-
 };
 
 &audio_clk_a {
@@ -337,14 +336,6 @@
 		 <&cpg CPG_MOD 723>,
 		 <&x13_clk>;
 	clock-names = "du.0", "du.1", "dclkin.0";
-
-	ports {
-		port@0 {
-			endpoint {
-				remote-endpoint = <&adv7123_in>;
-			};
-		};
-	};
 };
 
 &ehci0 {
@@ -512,10 +503,19 @@
 };
 
 &lvds1 {
+	status = "okay";
+
 	clocks = <&cpg CPG_MOD 727>,
-		 <&x13_clk>,
 		 <&extal_clk>;
-	clock-names = "fck", "dclkin.0", "extal";
+	clock-names = "fck", "extal";
+
+	ports {
+		port@1 {
+			lvds1_out: endpoint {
+				remote-endpoint = <&adv7123_in>;
+			};
+		};
+	};
 };
 
 &ohci0 {
-- 
2.17.1

